system-parachains/asset-hubs/asset-hub-polkadot/src/impls.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+
+// TODO: the types in the module copied from the PR: https://github.com/paritytech/polkadot-sdk/pull/3250
+// and should be removed when changes from the PR will get released.
+pub(crate) mod pool {
+	use super::*;
+	use core::marker::PhantomData;
+	use pallet_asset_conversion::PoolLocator;
+	use sp_core::Get;
+	use sp_runtime::traits::{TrailingZeroInput, TryConvert};
+
+	/// Pool locator that mandates the inclusion of the specified `FirstAsset` in every asset pair.
+	///
+	/// The `PoolId` is represented as a tuple of `AssetKind`s with `FirstAsset` always positioned
+	/// as the first element.
+	pub struct WithFirstAsset<FirstAsset, AccountId, AssetKind, AccountIdConverter>(
```

Comment:
what about the AssetHubKusama? It uses `pallet_asset_conversion::WithFirstAsset`, should we change it there also?

Reply:
only possible with migration, which will be available for the next release